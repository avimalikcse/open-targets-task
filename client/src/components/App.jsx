import * as React from "react";
import DiseaseAssociationContainer from "./features/diseaseAssociation/DiseaseAssociationContainer";
import Main from "./layouts/main";

export default function App() {

  return(
    <Main>
      <DiseaseAssociationContainer/>
    </Main>
  )
}