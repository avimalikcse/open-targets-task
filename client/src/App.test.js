import { render, screen, waitFor } from '@testing-library/react';
import App from './components/App';


test('renders App is working', async() => {
    render( < App / > );
    await waitFor(() => {
        const headerText = screen.getByText(/Genes association with Diseases/i);
        expect(headerText).toBeInTheDocument();
    })
});

test('Test snapshots of the components', async() => {
    const { asFragment } = render( < App / > );
    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    })
});
test('Test if layout Loaded', async() => {
    const { getByTestId } = render( < App / > );
    await waitFor(() => {
        expect(getByTestId("main-layout-testId")).toBeTruthy();
    })
});