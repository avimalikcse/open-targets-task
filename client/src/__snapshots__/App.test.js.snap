// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshots of the components 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1vazdj7-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-2l1qg7-MuiTypography-root"
      >
        Genes association with Diseases
      </h6>
    </div>
  </header>
  <main
    data-testid="main-layout-testId"
  >
    <div
      class="MuiBox-root css-10m0cap"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          data-testid="loader-testId"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </main>
  <footer
    class="MuiBox-root css-16ybtzv"
  >
    <p
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1xatcrf-MuiTypography-root"
    >
      All rights Reserved
    </p>
  </footer>
</DocumentFragment>
`;
